garage/spaces/box.py

```diff
@@ -26,10 +26,19 @@ def __init__(self, low, high, shape=None):
             self.low = low + np.zeros(shape)
             self.high = high + np.zeros(shape)
 
+        if dtype is None:
+            all_high = (high == 255).all() if shape is None else (high == 255)
```

Comment:
openai gym does this, auto-detect the dtype.
https://github.com/openai/gym/blob/master/gym/spaces/box.py

Reply:
Hmm. They do that, but they also throw a warning whenever dtype is None. I'm not convinced that we should duplicate this behavior. I guess either would be acceptable though, since we're aiming for "compatibility" with openai gym spaces.

If we do duplicate this behavior we should document why (linking to the openai gym implementation), and raise a warning (probably using the `warning` module).

Reply:
How about instead raising a warning when low=0, high=255 and dtype is not specified?

Reply:
Yeah, raising a warning in this case is a good idea. It's likely a mistake if low=0, high=255, and the `dtype` is not specified. Please also mention this in the docstring, as well as that it's different from the gym `Box`.

Reply:
Okay, it still doesn't do exactly what I would expect, but it's fully compatible with gym's Box, and I don't think it's likely to cause any problems. I'm satisfied.